packages/realm/src/app-services/SyncSession.ts

```diff
@@ -100,6 +106,84 @@ export function toBindingErrorHandler(onError: ErrorCallback) {
   };
 }
 
+/** @internal */
+export function toBindingErrorHandlerWithOnManual(
+  onError: ErrorCallback | undefined,
+  onManual: ClientResetFallbackCallback | undefined,
+) {
+  if (!onError && !onManual) {
+    throw new Error("need either onError or onManual");
+  }
+  if (onError && onManual) {
+    return (sessionInternal: binding.SyncSession, bindingError: binding.SyncError) => {
+      const session = new SyncSession(sessionInternal);
+      const error = fromBindingSyncError(bindingError);
+      if (error instanceof ClientResetError) {
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        onManual(session, error.config.path!);
+      } else {
+        onError(session, error);
+      }
+      session.resetInternal();
+    };
+  }
+  if (onError) {
+    return (sessionInternal: binding.SyncSession, bindingError: binding.SyncError) => {
+      const session = new SyncSession(sessionInternal);
+      const error = fromBindingSyncError(bindingError);
+      if (error instanceof ClientResetError) {
+        onError(session, error);
+      }
+      session.resetInternal();
+    };
+  }
+  // onManual must be true
+  return (sessionInternal: binding.SyncSession, bindingError: binding.SyncError) => {
+    const session = new SyncSession(sessionInternal);
+    const error = fromBindingSyncError(bindingError);
+    if (error instanceof ClientResetError) {
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      onManual!(session, error.config.path!);
```

Comment:
but confused here why TS can't prune onManual to be true after previous if statements

Reply:
Also it might be risky with `error.config.path!` is there a better way to fix this?

Reply:
I can't imagine a situation where `path` is either undefined or null.